pinot-core/src/main/java/com/linkedin/pinot/core/operator/MCombineGroupByOperator.java

```diff
@@ -138,35 +140,34 @@ public Block getNextBlock() {
    */
   private IntermediateResultsBlock combineBlocks()
       throws InterruptedException {
-    final int numOperators = _operators.size();
-    final IntermediateResultsBlock[] blocks = new IntermediateResultsBlock[numOperators];
+    int numOperators = _operators.size();
+    final IntermediateResultsBlock[] intermediateResultsBlocks = new IntermediateResultsBlock[numOperators];
     final CountDownLatch operatorLatch = new CountDownLatch(numOperators);
-
-    final List<AggregationInfo> aggregationsInfo = _brokerRequest.getAggregationsInfo();
-    final int numAggrFunctions = aggregationsInfo.size();
-
-    // TODO: use new aggregation functions to combine results.
-    final List<AggregationFunction> aggregationFunctions =
-        AggregationFunctionFactory.getAggregationFunction(_brokerRequest);
-
     final Map<String, Serializable[]> resultsMap = new ConcurrentHashMap<>();
 
+    List<AggregationInfo> aggregationInfos = _brokerRequest.getAggregationsInfo();
+    final AggregationFunctionContext[] aggregationFunctionContexts =
+        AggregationFunctionUtils.getAggregationFunctionContexts(aggregationInfos, null);
+    final int numAggregationFunctions = aggregationFunctionContexts.length;
+
     for (int i = 0; i < numOperators; i++) {
       final int index = i;
 
       _executorService.execute(new TraceRunnable() {
```

Comment:
I want to revisit this part in the combine operator clean up. The new group-by combine operator might not be ideal because it could occupy all threads available and block other queries. For this TraceRunnable, IMO, we'd better use a Callable and implement early termination when timeout happened. Will address in another PR.